### ISSUE TITLE
Disk Drive Sound

### ISSUE THREAD

**Issue by tomcw** [AppleWin/AppleWin#9]
**BerliOS Feature Request #2118**

**Date:** 2006-April-07 16:13
**Submitted By:** jimc
**Category:** none
**Assigned To:** none
**Priority:** 5
# 

Date: 2006-Apr-07 16:13
Sender: jimc
Logged In: YES 
user_id=27069
Browser: Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.1; SV1; .NET CLR 1.1.4322;
.NET CLR 2.0.50727)

There is a Virtual ][ emulator for MacOS that 
features a disk drive sound that's pretty accurate 
and really adds to the whole experience of 
emulation.  To me it makes a big difference and 
actually makes me FEEL like I'm using a real Apple 
II.  I've compaired loading sounds with the 
emulator and a real Apple and it's pretty much 
right-on... I'm crazy enough to have memorized 
quite a few booting sound patterns.... especially 
the Electronic Arts 'Clickity-Click'!  ..... Yeah I'm sad

Would something like that be possible with 
AppleWin as I feel that a feature that everyone 
would appreciate!

Thanks

**Comment by atylmo** [comment]
Has progress been made on this? It's something I'd love to see.

**Comment by tomcw** [comment]
No progress... I'll have to check if I ever got that patch & sound samples from jimc in 2008.

**Comment by atylmo** [comment]
Haha, I'm sorry. It's just something I've come to appreciate in computer emulation. The only open-source project I know of that does it for the Apple II is [OpenEmulator](https://github.com/openemulator/openemulator). It works well, but it's pretty much Mac-only and I don't know where to even start to port it.

I wonder if you could collaborate with them and share their samples?

**Comment by jvernet** [comment]
Virtual II also make disks noises

Le 18 sept. 2019 à 09:12, atylmo <notifications@github.com<mailto:notifications@github.com>> a écrit :


Haha, I'm sorry. It's just something I've come to appreciate in computer emulation. The only open-source project I know of that does it for the Apple II is OpenEmulator<https://github.com/openemulator/openemulator>. It works well, but it's pretty much Mac-only and I don't know where to even start to port it.

I wonder if you could collaborate with them and share their samples?

**Comment by atylmo** [comment]
Ah yeah, it does. They're lovely :) Maybe some sort of agreement could be hashed out there since it's not open source.

I also found these:https://github.com/ogoguel/activegs-ios/blob/master/Common.res/Apple_2_Disk_Drive_Sounds.txt. They seem like they could be a good fit for AppleWin.

**Comment by tomcw** [comment]
Here are the 6 .wav's from jimc's 2008 patch:

[diskii-sounds-jimc.zip](https://github.com/AppleWin/AppleWin/files/3883820/diskii-sounds-jimc.zip)

From the Disk.cpp patch: (about the 3 track sounds)
```
	x = abs(curTrack - lastTrack);
	if (x < 2) {Mix_PlayChannel(1, smtrack_snd, 0);} // one click
	if (x < 16 && x > 4) {Mix_PlayChannel(1, mtrack_snd, 0);} // medium sweep
	if (x > 16) {Mix_PlayChannel(1, track_snd, 0);} // full sweep
```

@atylmo: How do these compare with the github sounds you found?

**Comment by atylmo** [comment]
I like them! :) This set is more customizable for different patterns. The other sounded noticeably repetitious in the ActiveGS iOS app. I do quite like its spin-up sounds, though.

They sound wonderful to me! I haven't built it with the patch but I'm all for them.

**Comment by robinhood2014** [comment]
I know it's been a while since the last post on this issue, but I found out that MAME's Apple //c emulation recreates drive sounds nicely (if a bit inaudible). If we could perhaps fork those drive sounds into Applewin, that could be fantastic.

**Comment by robinhood2014** [comment]
Here are four ZIP files with the ROMs that MAME needs to emulate a //e. I should think one of them has the drive sounds we're looking for.
https://archive.org/download/mame-merged/mame-merged/a2diskiing.zip
https://archive.org/download/mame-merged/mame-merged/apple2e.zip (though we could just as easily use apple2.zip, etc.)
https://archive.org/download/mame-merged/mame-merged/d2fdc.zip
https://archive.org/download/mame-merged/mame-merged/votrax.zip (this probably isn't it, but I've included it here for completion's sake)

**Comment by robinhood2014** [comment]
The drive sounds we're looking for probably reside in this ZIP file. Inside it you'll find a file called "341-0027-a.p5"; feel free to make of that what you will.
https://archive.org/download/mame-merged/mame-merged/a2diskiing.zip

**Comment by xotmatrix** [comment]
@robinhood2014 That file is the ROM for the Disk II interface card. When you turn on your computer to boot a disk or type `PR#6`, that's the code that runs.

MAME's disk drive samples are here. They are licensed CC0.
https://github.com/mamedev/mame/tree/master/samples

**Comment by robinhood2014** [comment]
Ah yes. So the question becomes, how do we integrate those sounds into AppleWin? If necessary, we could include them in the zip file.

**Comment by TheShanMan** [comment]
Just came here wondering about this possibility in AppleWin. The Floppy Emu device has a relay based optional device to at least give it _some_ sound that corresponds to Disk II sounds. I thought man, I'd pay more for a device that plays audio samples of real drives. And then I wondered about AppleWin. Would love to see it here, though admittedly I don't use AppleWin all that much because I mostly use my real Apple IIe. But I think it would be a _really_ nice addition.

**Comment by xandark** [comment]
Yes, I'm following this thread eagerly too, it would be fantastic to have floppy disk sounds in AppleWin, it would complete the feel of the emulation.

**Comment by tomcw** [comment]
Yes, lots of continued interest in this (spanning nearly 2 decades now!), so thanks for continuing to put this back on our radar.
I agree it would be great to get this done.
